packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -174,7 +203,17 @@ function startClickAction(
         // If the clock is looking weird, just discard the click
         click.discard()
       } else {
-        click.stop(pageActivityEndEvent.hadActivity ? pageActivityEndEvent.end : undefined)
+        if (pageActivityEndEvent.hadActivity) {
+          click.stop(pageActivityEndEvent.end)
+        } else if (hadActivityOnPointerDown()) {
+          click.stop(
+            // using the click start as activity end, so the click will have some activity but its
+            // duration will be 0 (as the activity started before the click start)
+            click.startClocks.timeStamp
+          )
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏IIUC this case can happen if some activity starts at pointer down but no other activity starts at pointer up. 

- to handle correctly this particular case, it could be nice to not stop tracking the activity from pointer down since we could have pending requests
- more generally, should the action start always be at pointer down?

Reply:
> **💭 thought:** ‏IIUC this case can happen if some activity starts at pointer down but no other activity starts at pointer up.
> 
>     * to handle correctly this particular case, it could be nice to not stop tracking the activity from pointer down since we could have pending requests

Since we are not associating pending requests to the Click Action, I think it might be a bit confusing UI-wise to use them for the click duration. But we can talk about it


> 
>     * more generally, should the action start always be at pointer down?

As previously discussed in the RFC, this was discouraged, but we can still chalenge it. I think it would make this change more noticeable though.